src/github/api/mod.rs

```diff
@@ -83,17 +84,21 @@ pub async fn load_repositories(
                 ));
             }
 
-            let repo_state = create_repo_state(
+            let repo_state = match create_repo_state(
                 app.clone(),
                 installation_client.clone(),
                 team_api_client,
                 repo.clone(),
                 name.clone(),
             )
             .await
-            .map_err(|error| {
-                anyhow::anyhow!("Cannot load repository {:?}: {error:?}", repo.full_name)
-            });
+            {
+                Ok(repo_state) => repo_state,
```

Comment:
The complicated handling of results here was by design, because we wanted to have a hard error when the repos fail to load when the bot starts, but only log a warning when we reload the state during the bot's execution. By moving the error handling here, it is not possible to distinguish these two cases. So I would prefer keeping the previous approach of returning `Result<RepositoryState>`.